electrum/gui/qt/rbf_dialog.py

```diff
@@ -30,125 +33,110 @@ def __init__(
+        grid.addWidget(QLabel(_('New fee') + ':'), 2, 0)
+        grid.addLayout(self.fee_hbox, 2, 1, 1, 3)
+        grid.addWidget(HelpLabel(_("Fee target") + ": ", self.fee_combo.help_msg), 4, 0)
+        grid.addLayout(self.fee_target_hbox, 4, 1, 1, 3)
+        grid.setColumnStretch(4, 1)
+        # locktime
+        grid.addWidget(self.locktime_label, 5, 0)
+        grid.addWidget(self.locktime_e, 5, 1, 1, 2)
+        return grid
+
     def is_decrease_payment(self):
         return self.method_combo.currentIndex() == 1
 
     def set_decrease_payment(self):
         self.method_combo.setCurrentIndex(1)
 
-    def rbf_func(self, fee_rate) -> PartialTransaction:
-        raise NotImplementedError()  # implemented by subclasses
-
     def run(self) -> None:
         if not self.exec_():
             return
-        self.new_tx.set_rbf(True)
-        tx_label = self.wallet.get_label_for_txid(self.txid)
-        self.window.show_transaction(self.new_tx, tx_desc=tx_label)
-        # TODO maybe save tx_label as label for new tx??
-
-    def update(self):
+        if self.is_preview:
+            self.main_window.show_transaction(self.tx)
+            return
+        def sign_done(success):
+            if success:
+                self.main_window.broadcast_or_show(self.tx)
+        self.main_window.sign_tx(
+            self.tx,
+            callback=sign_done,
+            external_keypairs={})
+
+    def update_tx(self):
         fee_rate = self.feerate_e.get_amount()
         self._update_tx(fee_rate)
         self._update_message()
 
     def _update_tx(self, fee_rate):
         if fee_rate is None:
-            self.new_tx = None
+            self.tx = None
             self.message = ''
         elif fee_rate <= self.old_fee_rate:
-            self.new_tx = None
+            self.tx = None
             self.message = _("The new fee rate needs to be higher than the old fee rate.")
         else:
             try:
-                self.new_tx = self.rbf_func(fee_rate)
+                self.tx = self.make_tx(fee_rate)
             except CannotBumpFee as e:
-                self.new_tx = None
+                self.tx = None
                 self.message = str(e)
-        if not self.new_tx:
+        if not self.tx:
             return
-        delta = self.new_tx.get_fee() - self.tx.get_fee()
+        delta = self.tx.get_fee() - self.old_tx.get_fee()
         if not self.is_decrease_payment():
-            self.message = _("You will pay {} more.").format(self.window.format_amount_and_units(delta))
+            self.message = _("You will pay {} more.").format(self.main_window.format_amount_and_units(delta))
         else:
-            self.message = _("The recipient will receive {} less.").format(self.window.format_amount_and_units(delta))
+            self.message = _("The recipient will receive {} less.").format(self.main_window.format_amount_and_units(delta))
 
     def _update_message(self):
```

Comment:
Note that `self.message` is not visible anywhere in the dialog in practice, as it is being overwritten by `TxEditor.toggle_send_button`.
I guess `_BaseRBFDialog._update_message` is mostly redundant with `TxEditor.toggle_send_button` - maybe one could call or override the other.

Reply:
fixed in ad83eaaba19cc6f2a1f9418c5a1d9a4924b9149f